From ae891268caa25b15abf4a15d8eea23a93db7a18f Mon Sep 17 00:00:00 2001
From: rigaz29 <ryanbagas27@gmail.com>
Date: Wed, 30 Sep 2026 16:32:29 +0000
Subject: [PATCH] fs: susfs: update spoofed_blocks along with spoofed_size

update_sus_kstat takes the new file's size after a bind mount but kept
the original st_blocks. The susfs4ksu hide_vendor_sepolicy mounts then
report e.g. size 720622 with 1528 blocks (the original 781308-byte
file), where 720622 bytes need 1408. Take i_blocks from the new inode
too, as the older susfs 1.5 update path did.
---
 fs/susfs.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/fs/susfs.c b/fs/susfs.c
index dd6e98c6887..21ac45115fe 100644
--- a/fs/susfs.c
+++ b/fs/susfs.c
@@ -331,8 +331,10 @@ static int susfs_mark_inode_sus_kstat(char *target_pathname, struct st_susfs_sus
 			err = -ENOENT;
 			goto out_path_put_path;
 		}
-		if (is_update)
+		if (is_update) {
 			new_entry->info.spoofed_size = d_backing_inode(path.dentry)->i_size;
+			new_entry->info.spoofed_blocks = d_backing_inode(path.dentry)->i_blocks;
+		}
 
 		new_entry->is_fuse = true;
 		new_entry->target_dev = fi->inode.i_sb->s_dev;
@@ -350,8 +352,10 @@ static int susfs_mark_inode_sus_kstat(char *target_pathname, struct st_susfs_sus
 		goto out_path_put_path;
 	}
 
-	if (is_update)
+	if (is_update) {
 		new_entry->info.spoofed_size = d_backing_inode(path.dentry)->i_size;
+		new_entry->info.spoofed_blocks = d_backing_inode(path.dentry)->i_blocks;
+	}
 
 	new_entry->is_fuse = false;
 	new_entry->target_dev = inode->i_sb->s_dev;
-- 
2.43.0

